BlobForwarder/index.js

```diff
@@ -42,29 +42,51 @@ module.exports = async function main(context, myBlob) {
     context.log.warn('logs format is invalid');
     return;
   }
-  let compressedPayload;
-  let payloads = generatePayloads(buffer, context);
-  for (const payload of payloads) {
-    try {
-      compressedPayload = await compressData(JSON.stringify(payload));
-      try {
-        await retryMax(httpSend, NR_MAX_RETRIES, NR_RETRY_INTERVAL, [
+  let logLines = appendMetaDataToAllLogLines(buffer);
+  await compressAndSend(logLines, context);
+};
+
+function compressAndSend(data, context) {
+  return compressData(JSON.stringify(getPayload(data, context)))
+    .then((compressedPayload) => {
+      if (compressedPayload.length > NR_MAX_PAYLOAD_SIZE) {
+        if (data.length === 1) {
```

Comment:
As we are now checking for 1 MB limit post compression, I feel this check should be post compressData and should not be in getPayload. The purpose of this check is to log and stop recursion when a single data size (post compression) is beyond 1MB limit. Hence, I feel right to have this here rather than moving it to getPayload. Please correct me if I miss anything.

Reply:
Got it!